### ISSUE TITLE
SMB/CIFS external storages - writing issue

### ISSUE THREAD

**Issue by S74HK9hV** [nextcloud/server#7898]
sorry... i'll try to fill in more detail when i get home... hopefully this is enough to make sense of for now.

my configuration was working for a long time, and all of a sudden in december, writing to external storages stopped working. the mounts still work, and i can list/read files, but any write operations stopped working

- issue is not with WebDAV as writing to the normal data directory is OK
- smb user has write permissions.... tested by manually mounting the folders and logging in with same credentials
- issue persists using either the mobile client, or browser
- i may have upgraded my smbd to smb v3 in december (can't remember)... but i set that back to v2.1 and i still can't write to storage

### Steps to reproduce
1. set up SMB/CIFS external share (mounting is successful)
2. upload a file to the SMB/CIFS folder

### Expected behaviour
File should upload to the folder

### Actual behaviour
On mobile client, i get a progress bar that goes to 100%, and then i get an error saying that file could not be uploaded

On the browser (Firefox 57) i get a progress bar that looks like it goes to 100%, and then "Internal Server error"

### Server configuration

**Operating system**: Raspbian

**Web server:** Apache/2.4.25 (Raspbian)

**Database:** mariadb  Ver 15.1 Distrib 10.1.23-MariaDB, for debian-linux-gnueabihf (armv7l) using readline 5.2

**PHP version:** PHP 7.0.27-0+deb9u1

**Nextcloud version:** 12.0.4

**Updated from an older Nextcloud/ownCloud or fresh install:**

**Where did you install Nextcloud from:** Website/plus online updates 

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```

</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - activity: 2.5.2
  - admin_audit: 1.2.0
  - bruteforcesettings: 1.0.3
  - calendar: 1.5.7
  - contacts: 2.0.1
  - dav: 1.3.0
  - federatedfilesharing: 1.2.0
  - files: 1.7.2
  - files_external: 1.3.0
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - firstrunwizard: 2.1
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - notes: 2.3.2
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - password_policy: 1.2.2
  - provisioning_api: 1.2.0
  - spreed: 2.0.2
  - twofactor_backupcodes: 1.1.1
  - twofactor_totp: 1.3.1
  - updatenotification: 1.2.0
  - workflowengine: 1.2.0
Disabled:
  - comments
  - encryption
  - federation
  - files_accesscontrol
  - files_pdfviewer
  - files_sharing
  - files_texteditor
  - files_videoplayer
  - gallery
  - nextcloud_announcements
  - serverinfo
  - sharebymail
  - survey_client
  - systemtags
  - theming
  - user_external
  - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "----",
            "example.com"
        ],
        "datadirectory": "\/var\/www\/nextcloud\/data",
        "htaccess.RewriteBase": "\/",
        "overwrite.cli.url": "http:\/\/example.com\/nextcloud",
        "dbtype": "mysql",
        "version": "12.0.4.3",
        "dbname": "nextcloud",
        "dbhost": "localhost",
        "dbport": "",
        "dbtableprefix": "oc_",
        "instanceid": "oc6vg8xtxqxd",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "maintenance": false,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "enable_previews": false,
        "theme": "",
        "log_type": "owncloud",
        "logfile": "nextcloud.log",
        "loglevel": 3,
        "logdateformat": "F d, Y H:i:s",
        "filesystem_check_changes": 1,
        "updater.release.channel": "stable"
    }
}


```
</details>

**Are you using external storage, if yes which one:** SMB

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** WebDAV

### Client configuration
**Browser:** Firefox 57

**Operating system:** Debian stretch

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Fatal | webdav | Sabre\DAV\Exception: Could not write file contents | January 16, 2018 13:16:58
    /var/www/nextcloud/apps/dav/lib/Connector/Sabre/Directory.php - line 151: OCA\DAV\Connector\Sabre\File->put(Resource id #8)
    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php - line 1096: OCA\DAV\Connector\Sabre\Directory->createFile('test.txt', Resource id #8)
    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php - line 525: Sabre\DAV\Server->createFile('Photos/test.txt', Resource id #8, NULL)
    [internal function] Sabre\DAV\CorePlugin->httpPut(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
    /var/www/nextcloud/3rdparty/sabre/event/lib/EventEmitterTrait.php - line 105: call_user_func_array(Array, Array)
    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php - line 479: Sabre\Event\EventEmitter->emit('method PUT', Array)
    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php - line 254: Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
    /var/www/nextcloud/apps/dav/appinfo/v1/webdav.php - line 76: Sabre\DAV\Server->exec()
    /var/www/nextcloud/remote.php - line 162: require_once('/var/www/nextcl...')
    {main}

also, i got some weird output when i ran the occ file scan:
Error: Wrong parameters for Icewind\SMB\Exception\Exception([string $message [, long $code [, Throwable $previous = NULL]]]) in /var/www/nextcloud/apps/files_external/3rdparty/icewind/smb/src/Exception/Exception.php:17
Stack trace:
#0 /var/www/nextcloud/apps/files_external/3rdparty/icewind/smb/src/Exception/Exception.php(17): Exception->__construct('Unknown error (...', 'NT_STATUS_INVAL...')
#1 /var/www/nextcloud/apps/files_external/3rdparty/icewind/smb/src/Exception/Exception.php(35): Icewind\SMB\Exception\Exception::unknown('/@Recycle', 'NT_STATUS_INVAL...')
#2 /var/www/nextcloud/apps/files_external/3rdparty/icewind/smb/src/Parser.php(72): Icewind\SMB\Exception\Exception::fromMap(Array, 'NT_STATUS_INVAL...', '/@Recycle')
#3 /var/www/nextcloud/apps/files_external/3rdparty/icewind/smb/src/Share.php(392): Icewind\SMB\Parser->checkForError(Array, '/@Recycle')
#4 /var/www/nextcloud/apps/files_external/3rdparty/icewind/smb/src/Share.php(147): Icewind\SMB\Share->parseOutput(Array, '/@Recycle')
#5 /var/www/nextcloud/apps/files_external/lib/Lib/Storage/SMB.php(134): Icewind\SMB\Share->stat('/@Recycle')
#6 /var/www/nextcloud/apps/files_external/lib/Lib/Storage/SMB.php(419): OCA\Files_External\Lib\Storage\SMB->getFileInfo('/@Recycle')
#7 /var/www/nextcloud/lib/private/Files/Storage/Common.php(106): OCA\Files_External\Lib\Storage\SMB->filetype('@Recycle')
#8 /var/www/nextcloud/lib/private/Files/Storage/Common.php(140): OC\Files\Storage\Common->is_dir('@Recycle')
#9 /var/www/nextcloud/lib/private/Files/Storage/Common.php(160): OC\Files\Storage\Common->isCreatable('@Recycle')
#10 /var/www/nextcloud/lib/private/Files/Storage/Common.php(657): OC\Files\Storage\Common->getPermissions('@Recycle')
#11 /var/www/nextcloud/lib/private/Files/Storage/Wrapper/Wrapper.php(577): OC\Files\Storage\Common->getMetaData('@Recycle')
#12 /var/www/nextcloud/lib/private/Files/Storage/Wrapper/PermissionsMask.php(143): OC\Files\Storage\Wrapper\Wrapper->getMetaData('@Recycle')
#13 /var/www/nextcloud/lib/private/Files/Storage/Wrapper/Wrapper.php(577): OC\Files\Storage\Wrapper\PermissionsMask->getMetaData('@Recycle')
#14 /var/www/nextcloud/lib/private/Files/Storage/Wrapper/Availability.php(460): OC\Files\Storage\Wrapper\Wrapper->getMetaData('@Recycle')
#15 /var/www/nextcloud/lib/private/Files/Storage/Wrapper/Wrapper.php(577): OC\Files\Storage\Wrapper\Availability->getMetaData('@Recycle')
#16 /var/www/nextcloud/lib/private/Files/Cache/Scanner.php(113): OC\Files\Storage\Wrapper\Wrapper->getMetaData('@Recycle')
#17 /var/www/nextcloud/lib/private/Files/Cache/Scanner.php(152): OC\Files\Cache\Scanner->getData('@Recycle')
#18 /var/www/nextcloud/lib/private/Files/Cache/Scanner.php(422): OC\Files\Cache\Scanner->scanFile('@Recycle', 3, '2640', Object(OC\Files\Cache\CacheEntry), true)
#19 /var/www/nextcloud/lib/private/Files/Cache/Scanner.php(390): OC\Files\Cache\Scanner->handleChildren('', true, 3, '2640', true, 0)
#20 /var/www/nextcloud/lib/private/Files/Cache/Scanner.php(324): OC\Files\Cache\Scanner->scanChildren('', true, 3, '2640', true)
#21 /var/www/nextcloud/lib/private/Files/Utils/Scanner.php(242): OC\Files\Cache\Scanner->scan('', true, 3)
#22 /var/www/nextcloud/apps/files/lib/Command/Scan.php(161): OC\Files\Utils\Scanner->scan('/user')
#23 /var/www/nextcloud/apps/files/lib/Command/Scan.php(234): OCA\Files\Command\Scan->scanFiles('user', '/user', false, Object(Symfony\Component\Console\Output\ConsoleOutput), false)
#24 /var/www/nextcloud/3rdparty/symfony/console/Command/Command.php(256): OCA\Files\Command\Scan->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#25 /var/www/nextcloud/core/Command/Base.php(161): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#26 /var/www/nextcloud/3rdparty/symfony/console/Application.php(818): OC\Core\Command\Base->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#27 /var/www/nextcloud/3rdparty/symfony/console/Application.php(186): Symfony\Component\Console\Application->doRunCommand(Object(OCA\Files\Command\Scan), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#28 /var/www/nextcloud/3rdparty/symfony/console/Application.php(117): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#29 /var/www/nextcloud/lib/private/Console/Application.php(170): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#30 /var/www/nextcloud/console.php(100): OC\Console\Application->run()
#31 /var/www/nextcloud/occ(11): require_once('/var/www/nextcl...')
 

```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by S74HK9hV** [comment]
seems to be fixed after 12.0.5 upgrade... 

closing

**Comment by S74HK9hV** [comment]
seems to be fixed after 12.0.5 upgrade... 

closing
